### ISSUE TITLE
Invalid shift detection for a64 add/sub imm seems a bit roundabout

### ISSUE THREAD

**Issue by mstange** [iximeow/yaxpeax-arm#4]
Just something small I noticed while studying the code:

Here we bail out for `shift` values 0b10 and 0b11: https://github.com/iximeow/yaxpeax-arm/blob/e0fbc6335885d5363d823d5435994d2cfc8a19cb/src/armv8/a64.rs#L7943-L7954

`shift` is bits 22 and 23 of `word`: https://github.com/iximeow/yaxpeax-arm/blob/e0fbc6335885d5363d823d5435994d2cfc8a19cb/src/armv8/a64.rs#L7899

So if bit 23 of `word` is set, the shift is invalid.

`group_bits` is bits 23..26 of `word`: https://github.com/iximeow/yaxpeax-arm/blob/e0fbc6335885d5363d823d5435994d2cfc8a19cb/src/armv8/a64.rs#L7866

Here's the `group_bits` match arm that takes us into the add/sub imm branch: https://github.com/iximeow/yaxpeax-arm/blob/e0fbc6335885d5363d823d5435994d2cfc8a19cb/src/armv8/a64.rs#L7893-L7895

In other words, in the `0b011` case we already know that the shift is invalid.

Caveat: I have not read the instruction manual, so if the current code maps better to the spec, then it's probably worth keeping the structure as is.

**Comment by iximeow** [comment]
yep, this is a quirk that definitely fell out of a literal reading of the manual. it turns out that somewhere around the introduction of armv8.5 this got adjusted and is more obviously not a useful reading; `sh` is discussed as one bit (22) now, and the `0b011` case is used for `Add/subtract (immediate, with tags)` aka `addg` and `subg` instructions.

i'd never seen those before, and i assume that because differential testing didn't show those as incorrectly rejected, the version of capstone i'd tested against also doesn't know about them either?

**Comment by mstange** [comment]
Interesting. Thanks for the response!
